provider/bidengine/provider_attributes.go

```diff
@@ -88,6 +141,23 @@ loop:
 				pass.completeAllPending(result.auditor, result.providerAttr)
 			}
 			delete(pass.pending, result.auditor)
+		case <-pass.fetchattr:
+			pass.tryFetchAttributes(ctx)
+		case req := <-pass.reqattr:
+			pass.processAttrReq(ctx, req)
+		case <-pass.pushattr:
+			select {
+			case pass.currattr <- pass.attr:
+			default:
+			}
+		case attr := <-pass.newattr:
+			pass.attr = attr
+			pass.pushCurrAttributes()
+		case <-pass.errfetchattr:
+			// if attributes fetch fails give it retry withing reasonable timeout
+			time.AfterFunc(attrFetchRetryPeriod, func() {
+				pass.fetchAttributes()
```

Comment:
I'm trying to figure out what the utility of this `.fetchAttributes` method is? It looks like the same goroutine that receives from `pass.fetchattr` also calls `pass.fetchAttributes`. Couldn't you just use a boolean flag to indicate something needed to be done?